homeassistant/components/seventeentrack/config_flow.py

```diff
@@ -0,0 +1,153 @@
+        errors: dict[str, str] = {}
+        try:
+            client = await get_client(self.hass, user_input)
+            self.account_id = client.profile.account_id
+
+        except AuthenticationError:
+            errors["base"] = "invalid_auth"
+        except SeventeenTrackError:
+            errors["base"] = "cannot_connect"
+
+        return errors
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+
+            if not (errors := await self._async_validate_input(user_input)):
+                await self.async_set_unique_id(self.account_id)
+                self._abort_if_unique_id_configured()
+
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=CONFIG_SCHEMA,
+            errors=errors,
+        )
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        errors: dict[str, str] = {}
+        existing_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        if user_input is not None and existing_entry is not None:
+            if not (errors := await self._async_validate_input(user_input)):
+                self.hass.config_entries.async_update_entry(
+                    existing_entry,
+                    data={
+                        **existing_entry.data,
+                        **user_input,
+                    },
+                )
+                await self.hass.config_entries.async_reload(existing_entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_TOKEN): str,
+                }
+            ),
+            errors=errors,
+        )
+
+
+class SeventeenTrackOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle SeventeenTrack options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None) -> FlowResult:
+        """Manage options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        options = {
+            vol.Optional(
+                CONF_SCAN_INTERVAL,
```

Comment:
We don't allow scan interval as option. Every integration that supports config entries allows the user to turn off automatic polling and then the user can automate the update interval as needed with the entity service `homeassistant.update_entity`.